main.cpp

```diff
@@ -57,28 +77,36 @@ struct List {
 
     int pop_front() {
         int ret = head->value;
-        head = head->next;
+        head = std::move(head->next);
         return ret;
     }
 
     void push_front(int value) {
-        auto node = std::make_shared<Node>(value);
-        node->next = head;
-        if (head)
-            head->prev = node;
-        head = node;
+        auto node = std::unique_ptr<Node>(new Node(value));
+        node->next = std::move(head);
+		if (node->next)
+		{
+			node->next->prev = node.get();
+		}
+		head = std::move(node);
     }
 
     Node *at(size_t index) const {
         auto curr = front();
         for (size_t i = 0; i < index; i++) {
-            curr = curr->next.get();
+			if (!curr)
+			{
+				printf("Error:Exceed List Volumn!/n");
+				return NULL;
+			}	
+			curr = curr->next.get();
+			
         }
         return curr;
     }
 };
 
-void print(List lst) {  // 有什么值得改进的？
+void print(List lst) {  
```

Comment:
```suggestion
void print(List const &lst) {  
```
使用常引用，避免调用 List 的拷贝构造函数。